src/commands.ts

```diff
@@ -682,6 +834,7 @@ function updateAfterError(result: boolean, folderContext: FolderContext) {
  */
 export function register(ctx: WorkspaceContext) {
     ctx.subscriptions.push(
+        vscode.commands.registerCommand("swift.createProject", () => createProject(ctx)),
```

Comment:
createProject() has a few instances where it returns undefined. What's the experience for the user in these cases? Should createProject be providing guidance/errors to the user in the form of notifications?

Reply:
The places where createProject() returns undefined correspond to the user cancelling dialogs. I don't think we need any notifications in this case.

Reply:
👍 What's the behaviour from vscode for exceptions (e.g. the execSwift or mkdir fail)

Reply:
It pops up a modal with the error message from the exception.

Reply:
Have you verified what the message is for say `mkdir` failing? Would be good to make sure it is human readable.

Reply:
Good question! I tried it out myself and it looks like the error message is fairly self explanatory:

<img width="268" alt="Screenshot 2024-04-22 at 1 05 31 PM" src="https://github.com/swift-server/vscode-swift/assets/65295938/db2af8b2-43ab-4901-9aab-96d9b9ab632c">